vwifi.c

```diff
@@ -1898,13 +1906,45 @@ static struct wiphy *vwifi_cfg80211_add(void)
     struct wiphy *wiphy = NULL;
     enum nl80211_band band;
 
+    /* Inorder to customize vwifi's wiphy name, we use vwifi_wiphy_counter to
+     * keep track of the number of wiphy in vwifi, and use vwifi_wiphy_idx to
+     * retreive the value of vwifi_wiphy_counter.
+     */
+    int vwifi_wiphy_idx = atomic_inc_return(&vwifi_wiphy_counter);
+
+    /* atomic_inc_return makes it start at 1, make it start at 0 */
+    vwifi_wiphy_idx--;
+    if (unlikely(vwifi_wiphy_idx < 0)) {
+        atomic_dec(&vwifi_wiphy_counter);
+        return NULL;
+    }
+
     /* allocate wiphy context. It is possible just to use wiphy_new().
      * wiphy should represent physical FullMAC wireless device. We need
      * to implement add_virtual_intf() from cfg80211_ops for adding
      * interface(s) on top of a wiphy.
      * NULL means use the default phy%d naming.
+     * vwifi_wiphy_name is the custom-made vw_phy%d naming we use for
+     * wiphy in vwifi.
      */
-    wiphy = wiphy_new_nm(&vwifi_cfg_ops, 0, NULL);
+
+    /* Reference:
+     * Linux Kernel Tag : v6.7.1
+     * The default phy%d naming for wiphy in linux kernel depends on the value
```

Comment:
You don't have to mention Linux kernel since this module is intended to be specific to it.